include/swift/AST/ModuleDependencies.h

```diff
@@ -348,8 +348,14 @@ class SwiftBinaryModuleDependencyStorage : public ModuleDependencyInfoStorageBas
   /// The path to the .swiftSourceInfo file.
   const std::string sourceInfoPath;
 
-  /// The paths of all the .pch dependencies of this module.
-  const std::vector<std::string> preCompiledBridgingHeaderPaths;
+  /// The paths of all the .h dependencies of this module.
+  const std::vector<std::string> headerImports;
```

Comment:
Should this just be `std::string` or it is really possible to get multiple headerImports?

Reply:
https://github.com/apple/swift/blob/afeef855135363fc576bd172589e70f137c8f173/lib/Serialization/SerializedModuleLoader.cpp#L416
It's based on code in serialization which does allow any number of header dependencies.

Reply:
I understand the confusion, but I don't really see a reason to have this part of the scanner not match the binary module loader.

Reply:
My understanding is that the files in binary module should be your `headerSourceFiles` list? Is the bridging header always the first one? I think we either do:
```
std::string headerImports; // this is bridging header.
std::vector<std::string> headerSourceFiles; // this is all the textual headers from bridging header, include bridging header.
```
Or
```
std::vector<std::string> headerSourceFiles; // all the textual header include and bridging header is the first one.
```
If we go down the second route, maybe also verify that the dependency scanning result of the first file indeed returns a list of file that matches the list?

Reply:
No, `headerImports` **is** the files in the binary module. There are no guarantees in the format that the first file is *the* bridging header. `headerSourceFiles` is queried from the `TranslationUnitDeps::FileDeps` that we get from the Clang scanner action on a given header we ask it to scan for us.

Reply:
Realistically, we should consolidate this to be just one header dependency, and perhaps error out when we detect more than one header input on a binary module. Then we can consolidate the code to be:
```
std::string headerImports; // this is bridging header.
std::vector<std::string> headerSourceFiles;
```

Reply:
Refactored the code to follow this assumption and added an assert on reading out more than one header dependency.